invenio/modules/search/testsuite/test_elasticsearch_walker.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Unit tests for the elasticsearch AST walker"""
+
+from invenio.testsuite import make_test_suite, run_test_suite, InvenioTestCase
+from invenio_query_parser.ast import (
+    AndOp, KeywordOp, OrOp, NotOp, Keyword, Value, SingleQuotedValue,
+    DoubleQuotedValue, ValueQuery, RegexValue, RangeOp, EmptyQuery,
+    GreaterOp, GreaterEqualOp, LowerOp, LowerEqualOp
+)
+
+from invenio_query_parser.contrib.spires.ast import SpiresOp
+from invenio.modules.search.walkers.elasticsearch import ElasticSearchDSL
+
+
+class TestElasticSearchWalker(InvenioTestCase):
+
+    """Test transformations from the query-parser produced AST to the
+       elasticsearch DSL query
+    """
+
+    def setUp(self):
+        self.converter = ElasticSearchDSL()
+        self.converter.keyword_dict = {"foo": ["test1", "test2"]}
+
+    # Empty query
+    def test_empty_query(self):
+        self.assertEqual(EmptyQuery("").accept(self.converter),
+                         {"match_all": {}})
+
+    # Value queries
+    def test_value_query(self):
+        self.assertEqual(ValueQuery(Value("bar")).accept(self.converter),
+                         {"multi_match":
```

Comment:
@marioskogias please try to avoid writing this:
```python
long_function_name(some_long_argument,
                   {"key":
                       {...}})
```
but rather write something like
```python
long_function_name(some_long_argument, {
    "key": {
        ...
    }
})
```